packages/near-membrane-base/src/environment.ts

```diff
@@ -164,10 +164,14 @@ export class VirtualEnvironment {
         this.redCallableLinkPointers = redCallableLinkPointers;
     }
 
-    evaluate(sourceText: string): void {
+    evaluate(sourceText: string): any {
         try {
-            this.redCallableEvaluate(sourceText);
-        } catch (e) {
+            const result = this.redCallableEvaluate(sourceText);
+            if (typeof result === 'object' && result !== null) {
+                throw new TypeError();
+            }
```

Comment:
@caridy Halp! ☝️ I'm not really sure what's needed to get the return value from `realm.evaluate` or the fact that when we have a realm it will still be running the serialized `init` function within it which will have a `redCallableEvaluate` callback I assume. Does something need to be done. What I have in place above is not correct.